### PR TITLE
add mongodb-memory-server

### PR DESCRIPTION
**Pull request by Uzlopak**
adding mongodb-memory-server

some tests fail, but i dont know how to fix them.

### CODE REVIEW

File: test/agenda.js [hunk 1]
```diff
@@ -6,11 +6,16 @@ const delay = require('delay');
 const Job = require('../lib/job');
 const hasMongoProtocol = require('../lib/agenda/has-mongo-protocol');
 const Agenda = require('..');
+const mongoServer = require('./mongo-server');
+
+let mongoCfg;
+// eslint-disable-next-line no-undef
```
Comment (joeframbach) [inline test/agenda.js:12]: Add "before" to the globals list on line 1.

**Comment by Uzlopak** [comment]
failing tests:

- "should run immediately if options.skipImmediate is false"
- all unique tests

**Comment by Uzlopak** [comment]
@joeframbach 

any thoughts?

**Comment by pdfowler** [comment]
@Uzlopak I have build on your work and submitted my own PR which addresses the outstanding issues in this one. LMK if you have any questions

**Pull request closed without merging**